src/main/java/com/google/googleidentity/security/LoginServlet.java

```diff
@@ -0,0 +1,76 @@
+package com.google.googleidentity.security;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.Singleton;
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import freemarker.template.Version;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.Map;
+
+@Singleton
+public class LoginServlet extends HttpServlet {
+
+    private static final long serialVersionUID = 1L;
+
+    @Inject
+    private Provider<UserSession> session = null;
+
+    private Configuration configuration;
+
+    public void init() throws ServletException {
+        Version version= new Version("2.3.30");
+
+        configuration = new Configuration(version);
+
+        configuration.setServletContextForTemplateLoading(getServletContext(), "template");
+    }
+
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        try {
+            LoginPage(request,  response);
+        } catch (TemplateException e) {
+            e.printStackTrace();
```

Comment:
Log the exception via java.util.logging.Logger.

App engine docs: https://cloud.google.com/appengine/docs/standard/java/logs#writing_application_logs

WARNING level is probably good. Logger has an overload that accepts a Throwable:
https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html#log(java.util.logging.Level,%20java.lang.String,%20java.lang.Throwable)